src/OpenSage.Game/Data/Apt/Geometry.cs

```diff
@@ -17,13 +43,273 @@ public static Geometry FromFileSystemEntry(FileSystemEntry entry)
+                        if (line.V1.X < topLeft.X) topLeft.X = line.V1.X;
+                        if (line.V0.Y < topLeft.Y) topLeft.Y = line.V0.Y;
+                        if (line.V1.Y < topLeft.Y) topLeft.Y = line.V1.Y;
+
+                        if (line.V0.X > botRight.X) botRight.X = line.V0.X;
+                        if (line.V1.X > botRight.X) botRight.X = line.V1.X;
+                        if (line.V0.Y > botRight.Y) botRight.Y = line.V0.Y;
+                        if (line.V1.Y > botRight.Y) botRight.Y = line.V1.Y;
+                    }
+                }
+                else if (entry is GeometrySolidTriangles gst)
+                {
+                    foreach (var tri in gst.Triangles)
+                    {
+                        if (tri.V0.X < topLeft.X) topLeft.X = tri.V0.X;
+                        if (tri.V1.X < topLeft.X) topLeft.X = tri.V1.X;
+                        if (tri.V2.X < topLeft.X) topLeft.X = tri.V2.X;
+                        if (tri.V0.Y < topLeft.Y) topLeft.Y = tri.V0.Y;
+                        if (tri.V1.Y < topLeft.Y) topLeft.Y = tri.V1.Y;
+                        if (tri.V2.Y < topLeft.Y) topLeft.Y = tri.V2.Y;
+
+                        if (tri.V0.X > botRight.X) botRight.X = tri.V0.X;
+                        if (tri.V1.X > botRight.X) botRight.X = tri.V1.X;
+                        if (tri.V2.X > botRight.X) botRight.X = tri.V2.X;
+                        if (tri.V0.Y > botRight.Y) botRight.Y = tri.V0.Y;
+                        if (tri.V1.Y > botRight.Y) botRight.Y = tri.V1.Y;
+                        if (tri.V2.Y > botRight.Y) botRight.Y = tri.V2.Y;
+                    }
+                }
+                else if (entry is GeometryTexturedTriangles gtt)
+                {
+                    foreach (var tri in gtt.Triangles)
+                    {
+                        if (tri.V0.X < topLeft.X) topLeft.X = tri.V0.X;
+                        if (tri.V1.X < topLeft.X) topLeft.X = tri.V1.X;
+                        if (tri.V2.X < topLeft.X) topLeft.X = tri.V2.X;
+                        if (tri.V0.Y < topLeft.Y) topLeft.Y = tri.V0.Y;
+                        if (tri.V1.Y < topLeft.Y) topLeft.Y = tri.V1.Y;
+                        if (tri.V2.Y < topLeft.Y) topLeft.Y = tri.V2.Y;
+
+                        if (tri.V0.X > botRight.X) botRight.X = tri.V0.X;
+                        if (tri.V1.X > botRight.X) botRight.X = tri.V1.X;
+                        if (tri.V2.X > botRight.X) botRight.X = tri.V2.X;
+                        if (tri.V0.Y > botRight.Y) botRight.Y = tri.V0.Y;
+                        if (tri.V1.Y > botRight.Y) botRight.Y = tri.V1.Y;
+                        if (tri.V2.Y > botRight.Y) botRight.Y = tri.V2.Y;
+                    }
+                }
+            }
+
+            var size = botRight - topLeft;
+            BoundingBox = new RectangleF(topLeft.X, topLeft.Y, size.X, size.Y);
+        }
+    }
+
+    public struct GeometrySolidTriangles : GeometryEntry
```

Comment:
Any particular reason why this is a `struct`? Usually if a type involves inheritance, and reference fields (like the `List<Triangle2D>` below) then I would make it a `class`.

Reply:
I thought struct's might be faster/smaller/better. Will change